internal/common/server_status.go

```diff
@@ -11,24 +11,33 @@ import (
 	"github.com/edulinq/autograder/internal/util"
 )
 
+type ServerInitiator string
+
 const (
-	PID_SOCK_LOCK                  = "internal.common.PID_SOCK_LOCK"
+	SERVER_STATUS_LOCK             = "internal.common.SERVER_STATUS_LOCK"
 	STATUS_FILENAME                = "status.json"
 	UNIX_SOCKET_RANDNUM_SIZE_BYTES = 32
 )
 
+const (
+	PRIMARY_SERVER  ServerInitiator = "primary-server"
+	CMD_SERVER      ServerInitiator = "cmd-server"
+	CMD_TEST_SERVER ServerInitiator = "cmd-test-server"
+)
+
 type StatusInfo struct {
-	Pid            int    `json:"pid"`
-	UnixSocketPath string `json:"unix_socket_path"`
+	Pid             int             `json:"pid"`
+	UnixSocketPath  string          `json:"unix-socket-path"`
+	ServerInitiator ServerInitiator `json:"server-initiator"`
```

Comment:
Just call it "Initiator".

You already call it that elsewhere, and we try to avoid variables/members with the same names as types (if possible).